bears/c_languages/codeclone_detection/ClangComplexityBear.py

```diff
@@ -0,0 +1,88 @@
+                                considered to be normal. The value of 10 had
+                                received substantial corroborating evidence.
+                                But the general recommendation: "For each
+                                module, either limit cyclomatic complexity to
+                                [the agreed-upon limit] or provide a written
+                                explanation of why the limit was exceeded."
+        """
+        self.complexities = {}
+        counter = {'decisions': 0, 'exits': 0}
+
+        def _traverse(cursor, filename, top_function_level=False):
+            """
+            Traversing clang AST to calculate cyclomatic complexity.
+
+            :param cursor:             The node to print. (Clang cursor.)
+            :param filename:           Name of the file being processed.
+            :param top_function_level: Whether cursor is in the top level of
+                                       the function.
+            """
+            file = cursor.location.file
+
+            if file is not None and file.name == filename:
+                cursor_kind = cursor.kind
+            else:
+                cursor_kind = None
+
+            for child in cursor.get_children():
+                if cursor_kind == CursorKind.FUNCTION_DECL:
+                    if child.kind == CursorKind.PARM_DECL:
+                        continue
+                    counter['exits'] = counter['decisions'] = 0
+                    _traverse(child, filename, top_function_level=True)
+                    line = cursor.extent.start.line
+                    self.complexities[cursor.displayname] = (
+                        line, counter['decisions'], counter['exits'])
+                    return
+                if child.kind in self.decisive_cursor_kinds:
+                    counter['decisions'] += 1
+                elif child.kind == CursorKind.RETURN_STMT:
+                    counter['exits'] += 1
+                    if top_function_level:
+                        # There is no point to move forward, so just return.
+                        return
+                _traverse(child, filename)
+
+            if top_function_level:
+                # If return statement was avoided.
+                counter['exits'] += 1
+
+        root = Index.create().parse(filename).cursor
+        _traverse(root, filename)
+
+        for func_name, (line, dec_pts, exit_pts) in self.complexities.items():
+            complexity = max(1, dec_pts - exit_pts + 2)
+            if complexity > max_complexity:
+                yield Result.from_values(
+                    self,
+                    "The cyclomatic complexity of function {function} "
+                    "exceeded maximal recommended value of {rec_value}."
+                    .format(
+                        function=func_name,
+                        rec_value=max_complexity,
+                    ),
```

Comment:
we usually close brackets right after the last element, we're using PEP8 so you're free to do it this way also